nanomq/pub_handler.c

```diff
@@ -91,15 +91,15 @@ static void
 foreach_client(
     uint32_t *cli_ctx_list, nano_work *pub_work, struct pipe_content *pipe_ct)
 {
-	bool     equal = false;
 	uint32_t pids;
 	int      ctx_list_len;
 
+	ctx_list_len = cvector_size(cli_ctx_list);
+
 	// Dont using msg info buf, Just for Cheat Compiler
 	mqtt_msg_info *msg_info, msg_info_buf;
 	cvector(mqtt_msg_info) msg_infos = pipe_ct->msg_infos;
-
-	ctx_list_len = cvector_size(cli_ctx_list);
+	cvector_grow(msg_infos, ctx_list_len);
```

Comment:
kindly ask what bug have you found here?

Reply:
> kindly ask what bug have you found here?

cvector would grow (realloc) several times under high throughput, which reduces performance significantly.